### PR TITLE
Adding QC scripts for the Bisulfite pipeline

### PR DESCRIPTION
**Pull request by saimukund20**

### CODE REVIEW

File: Bisulfite_QC_CpGretentiondistribution.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  if [[ `which awk 2>&1 >/dev/null` ]]; then echo "awk does not exist in PATH"; exit 1; fi
+  for var in BISCUIT_CPGBED BISCUIT_CGIBED BISCUIT_RMSK BISCUIT_EXON BISCUIT_GENE BISCUIT_TOPGC_BED BISCUIT_BOTGC_BED input_bam input_vcf; do
+    if [[ ${!var} != "<unset>" && ! -f ${!var} ]]; then
+      >&2 echo "$var: ${!var} does not exist."
+      exit 1;
+    fi
+  done
+
+  echo "Running."
+  set -xe
```
Comment (chrisamiller) [inline Bisulfite_QC_CpGretentiondistribution.sh]: Probably not a bad idea to make this  `set -xeou pipefail` instead.  That way, if one of the steps that gets piped to another step fails, the whole thing will die.  As is, there are a couple of places where biscuit or bedtools could fail and downstream steps could succeed, and the workflow would succeed, but with weird or truncated results (this applies to all the QC bash scripts)

File: Bisulfite_QC_mappingsummary.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    samtools view -F 0x100 -f 0x4 $input_bam | wc -l | cat <(echo -ne "unmapped\t") - >>mapq_table.txt
+    samtools view -F 0x104 $input_bam | awk '{cnt[$5]+=1}END{for(mapq in cnt) {print mapq"\t"cnt[mapq];}}' | sort -k1,1n >>mapq_table.txt
+  fi
+
+}
+
+input_vcf="<unset>"
+
+function usage {
+  >&2 echo "Usage: Bisulfite_QC_mappingsummary.sh input_vcf input_bam referecnce_genome setup_file"
```
Comment (chrisamiller) [inline Bisulfite_QC_mappingsummary.sh]: s/referecnce/reference/  (for all scripts)

File: Bisulfite_QC_bisulfiteconversion.sh [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  ## bisulfite conversion
+  ##########################
+  [[ ! -f "$input_vcf" ]] && BISCUIT_QC_BSCONV=false
+  if [[ "$BISCUIT_QC_BSCONV" == true ]]; then
+    >&2 echo "`date`---- BISCUIT_QC_BSCONV ----"
+    echo -e "BISCUITqc Conversion Rate by Base Average Table" >totalBaseConversionRate.txt
+    biscuit vcf2bed -et c $input_vcf | awk '{beta_sum[$6]+=$8; beta_cnt[$6]+=1;} END{print "CA\tCC\tCG\tCT"; print beta_sum["CA"]/beta_cnt["CA"]"\t"beta_sum["CC"]/beta_cnt["CC"]"\t"beta_sum["CG"]/beta_cnt["CG"]"\t"beta_sum["CT"]/beta_cnt["CT"];}' >>totalBaseConversionRate.txt
+
+    echo -e "BISCUITqc Conversion Rate by Read Average Table" >totalReadConversionRate.txt
+    samtools view -hq 40 $input_bam | biscuit bsconv -b $BISCUIT_REFERENCE - | awk '{for(i=1;i<=8;++i) a[i]+=$i;}END{print "CpA\tCpC\tCpG\tCpT"; print a[1]/(a[1]+a[2])"\t"a[3]/(a[3]+a[4])"\t"a[5]/(a[5]+a[6])"\t"a[7]/(a[7]+a[8]);}' >>totalReadConversionRate.txt
```
Comment (chrisamiller) [inline Bisulfite_QC_bisulfiteconversion.sh:33]: This line and a few others limit to base quality 40. That's probably fine, just want to make sure it was intentional!

**Review by chrisamiller — CHANGES_REQUESTED** [review]

**Comment by saimukund20** [comment]
Made a new commit adding the new set command. Also spellchecked reference in all scripts.

**Comment by chrisamiller** [comment]
+1 - looks good

**Review by chrisamiller — APPROVED** [review]

**Pull request merged**